rpgpy/spcutil.py

```diff
@@ -0,0 +1,150 @@
+    mask2D = np.all(mask, axis=2)
+    ranges = np.append(LV0meta['RngOffs'], LV0meta['RAltN'])
+
+    for iC in range(LV0meta['SequN']):
+        Dopp_res = np.mean(np.diff(LV0meta['velocity_vectors'][iC]))
+
+        for iR in range(ranges[iC], ranges[iC + 1]):
+            for iT in range(n_ts):
+                if mask2D[iT, iR]: continue
+                _, (lb, rb) = find_peak_edges(spec_lin[iT, iR, :])
+                moments[iT, iR, :] = radar_moment_calculation(
+                    spec_lin[iT, iR, lb:rb],
+                    LV0meta['velocity_vectors'][iC][lb:rb]
+                )
+
+        # shift mean Doppler velocity by half a bin
+        moments[:, ranges[iC]:ranges[iC + 1], 1] -= Dopp_res / 2.0
+
+    # create the mask where invalid values have been encountered
+    invalid_mask = np.full((n_ts, n_rg), True)
+    invalid_mask[np.where(moments[:, :, 0] > 0.0)] = False
+
+    moments = {var: moments[:, :, i] for i, var in enumerate(['Ze', 'MeanVel', 'SpecWidth', 'Skewn', 'Kurt'])}
+
+    for mom in moments.keys():
+        moments[mom][invalid_mask] = -999.0
+
+    return moments
+
+
+@jit(nopython=True, fastmath=True)
+def radar_moment_calculation(signal: np.array, vel_bins: np.array) -> np.array:
+    """
+    Calculation of radar moments: reflectivity, mean Doppler velocity, spectral width,
+        skewness, and kurtosis of one Doppler spectrum. Optimized for the use of Numba.
+
+    Note:
+        Divide the signal_sum by 2 because vertical and horizontal channel are added.
+        Subtract half of of the Doppler resolution from mean Doppler velocity, because
+
+    Args:
+        - signal: detected signal from a Doppler spectrum
+        - vel_bins: extracted velocity bins of the signal (same length as signal)
+
+    Returns:
+        dict containing
+
+            - Ze_lin: reflectivity (0.Mom) over range of velocity bins [mm6/m3]
+            - VEL: mean velocity (1.Mom) over range of velocity bins [m/s]
+            - sw: spectrum width (2.Mom) over range of velocity bins [m/s]
+            - skew: skewness (3.Mom) over range of velocity bins
+            - kurt: kurtosis (4.Mom) over range of velocity bins
+    """
+
+    signal_sum = np.sum(signal)  # linear full spectrum Ze [mm^6/m^3], scalar
+    Ze_lin = signal_sum / 2.0
+    pwr_nrm = signal / signal_sum  # determine normalized power (NOT normalized by Vdop bins)
+
+    VEL = np.sum(vel_bins * pwr_nrm)
+    vel_diff = vel_bins - VEL
+    vel_diff2 = vel_diff * vel_diff
+    sw = np.sqrt(np.abs(np.sum(pwr_nrm * vel_diff2)))
+    sw2 = sw * sw
+    skew = np.sum(pwr_nrm * vel_diff * vel_diff2 / (sw * sw2))
+    kurt = np.sum(pwr_nrm * vel_diff2 * vel_diff2 / (sw2 * sw2))
+
+    return np.array((Ze_lin, VEL, sw, skew, kurt), dtype=float())
+
+
+@jit(nopython=True, fastmath=True)
+def find_peak_edges(signal: np.array, threshold: float = -1, imaxima: int = -1) -> Tuple[float, List[int]]:
```

Comment:
Do we need these optional arguments?

You should write unit test for this function.

Reply:
Handling jit during unit test is not that straight forward, I found some answers using environmental variables to disable during testing. Is this solution ok? I wrote a wrapper function and added 2 test cases.

Reply:
I can take a look at this.